### PR TITLE
add gulag emoji settings, modify emoji settings

### PR DESCRIPTION
**Pull request by AlexAppleManzer**
3 things really being done here

I refactored settings_cog to allow for you to make your own functions in response to the reaction add. I used this to allow for reacting to messages to get the setting for emoji's way easier.

I added the gulag_emoji setting because the work was basically already done.

I changed the gulag command slightly because it was bothering me.

### CODE REVIEW

File: shard/src/guild_settings.py [hunk 1]
```diff
@@ -91,6 +91,15 @@ def starboard_threshold(self, new_threshold: int):
         self._settings_dict['starboard_threshold'] = new_threshold
         self._update_db()
 
+    @property
+    def gulag_emoji(self) -> str:
+        return self._settings_dict['gulag_emoji'] if 'gulag_emoji' in self._settings_dict else "⚒️"
```
Comment (AlexAppleManzer) [inline shard/src/guild_settings.py:96]: this means that current servers will have to change their settings to use their custom gulag emoji's again

File: shard/src/ext/settings_cog.py [hunk 2]
```diff
@@ -342,6 +406,11 @@ async def roleids(self, ctx):
             lem.add(role.name, role.id)
         await ctx.channel.send(embed=lem.get_embed())
 
+    @commands.command(aliases=['listsettingstags', 'settingstags', 'stags'])
+    async def list_settings_tags(self, ctx):
+        '''Get the list of tags for the settings command'''
+        await ctx.channel.send(f"Tags for settings are: {self.list_tags()}")
```
Comment (AlexAppleManzer) [inline shard/src/ext/settings_cog.py:412]: not sure if this command is necessary...

File: shard/src/ext/settings_cog.py [hunk 3]
```diff
@@ -238,13 +254,44 @@ async def parse(self, ctx, msg, settings):
...
+        super().__init__(
+            "Gulag Emoji",
+            PICK,
+            'This is the emoji that is used to tally up gulag votes. '
+            'React to this message to modify it:',
+            'gulag_emoji',
+            tags=["gulag"])
+
+    async def parse(self, ctx, msg, settings):
+        return str(msg)
```
Comment (AlexAppleManzer) [inline shard/src/ext/settings_cog.py:268]: note: if you use a custom emoji from another guild you might break everything. I don't have nitro so i can't really test

File: shard/src/guild_settings.py (hunk 1 above)
Comment (johnyburd) [reply shard/src/guild_settings.py:96]: pretty sure about 3 server use it, so better now than later

File: shard/src/ext/settings_cog.py (hunk 2 above)
Comment (johnyburd) [reply shard/src/ext/settings_cog.py:412]: I like it, although the default name is a little clumsy

File: shard/src/ext/settings_cog.py (hunk 3 above)
Comment (johnyburd) [reply shard/src/ext/settings_cog.py:268]: it just pastes the string in. I think it's pretty obvious what's going on so I'm going to merge it like this

**Pull request merged**